WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubfilterListItem.kt

```diff
@@ -1,19 +1,107 @@
 package org.wordpress.android.ui.reader.subfilter
 
+import org.json.JSONObject
+import org.wordpress.android.R
+import org.wordpress.android.datasets.ReaderBlogTable
 import org.wordpress.android.models.ReaderBlog
 import org.wordpress.android.models.ReaderTag
+import org.wordpress.android.models.ReaderTagType
 import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.DIVIDER
 import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.SECTION_TITLE
 import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.SITE
 import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.SITE_ALL
 import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.TAG
+import org.wordpress.android.ui.reader.utils.ReaderUtils
 import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+
+const val JSON_KEY_ITEM_TYPE = "item-type"
+const val JSON_KEY_SITE_FEED_ID = "feed-id"
+const val JSON_KEY_SITE_BLOG_ID = "blog-id"
+const val JSON_KEY_TAG_NAME = "tag-name"
+const val JSON_KEY_TAG_TYPE = "tag-type"
 
 sealed class SubfilterListItem(val type: ItemType) {
     open var isSelected: Boolean = false
-    open val onClickAction: ((filter: SubfilterListItem) -> Unit)? = null
+    open var onClickAction: ((filter: SubfilterListItem) -> Unit)? = null
     open val label: UiString? = null
 
+    companion object {
+        @JvmStatic
+        fun fromJson(json: String): SubfilterListItem {
```

Comment:
What do you think about moving this logic into a separate injected class where it could be tested? Maybe create something like `SubfilterListItemMapper`? Actually can't you use a simpified data class (without inheritance, with nullable fields) and Gson? Using JSON object feels a little error prone.